mesh/mesh.cpp

```diff
@@ -12021,6 +12021,9 @@ FaceGeometricFactors::FaceGeometricFactors(const Mesh *mesh,
    IntRule = &ir;
    computed_factors = flags;
 
+   MemoryType d_mt = Device::Allows(Backend::RAJA_CUDA) == true ?
```

Comment:
We should use an agnostic way of using the non-default memory type, as suggested/done in the `geometricFactors`.

Reply:
@artv3, could you take a look at this suggestion ☝️

Reply:
Yes! Sorry for the delay, it is on my list.

Reply:
Thanks, just refactored it like in geometricFactors.